### ISSUE TITLE
Struct generated by "convert to struct" refactoring should implement "IEquatable<T>", operator== and operator !=

### ISSUE THREAD

**Issue by AndreasHeisel** [dotnet/roslyn#54767]
The refactoring "convert to struct" generates a struct replacement for value tuples. 

For example

`(int X, int Y)`

results in

```cs
internal struct NewStruct
{
	public int X;
	public int Y;

	public NewStruct(int x, int y)
	{
		X = x;
		Y = y;
	}

	public override bool Equals(object obj)
	{
		return obj is NewStruct other &&
					 X == other.X &&
					 Y == other.Y;
	}

	public override int GetHashCode()
	{
		return HashCode.Combine(X, Y);
	}

	public void Deconstruct(out int x, out int y)
	{
		x = X;
		y = Y;
	}

	public static implicit operator (int X, int Y)(NewStruct value)
	{
		return (value.X, value.Y);
	}

	public static implicit operator NewStruct((int X, int Y) value)
	{
		return new NewStruct(value.X, value.Y);
	}
}
```

To enable equality checking without boxing, I think it should implement IEquatable<T> this way:

```cs
internal struct NewStruct : IEquatable<NewStruct>
{
	...

	public bool Equals(NewStruct other)
	{
		return X == other.X && Y == other.Y;
	}

	public override bool Equals(object obj)
	{
		return obj is NewStruct other && Equals(other);
	}

	public static bool operator==(NewStruct x, NewStruct y)
	{
		return x.Equals(y);
	}

	public static bool operator !=(NewStruct x, NewStruct y)
	{
		return !x.Equals(y);
	}
	...
}
```

**Comment by CyrusNajmabadi** [comment]
Yes.  this is reasonable.  we would likely accept a PR here @AndreasHeisel if you're interested in contributing.  The code in quesiton is at: https://github.com/dotnet/roslyn/blob/2c03a468942ef441c96f0b8408c036b2fe39b044/src/Features/Core/Portable/ConvertTupleToStruct/AbstractConvertTupleToStructCodeRefactoringProvider.cs#L813-L834

Thanks!

**Comment by AndreasHeisel** [comment]
@CyrusNajmabadi : I tried and failed. 

It look's straightforward, but it isn't - at least for a newbie in this codebase. 

I found `IGenerateEqualsAndGetHashCodeService.GenerateIEquatableEqualsMethodAsync`, but it needs a `INamedTypeSymbol` of `IEquatable<NewStruct>`. I couldn't figure out how to create this.

Nevertheless I was not aware of the fact, that this refactoring creates a `record struct` in .NET 6. This should do the trick anyway.
